src/csharp/Grpc.Core/DeserializationContext.cs

```diff
@@ -41,6 +43,9 @@ public abstract class DeserializationContext
         /// (as there is no practical reason for doing so) and <c>DeserializationContext</c> implementations are free to assume so.
         /// </summary>
         /// <returns>byte array containing the entire payload.</returns>
-        public abstract byte[] PayloadAsNewBuffer();
+        public virtual byte[] PayloadAsNewBuffer()
+        {
+            throw new NotImplementedException();
```

Comment:
just wondering: why was this switched from `abstract` to having a default that's just unusable? Any `DeserializationContext` that doesn't implement this wouldn't work, no?

Reply:
This has been done with the idea that in the future more methods will be added (e.g. to access the payload with more efficiency) and when added, they will be added as virtual methods with a default implementation so that the backward compatibility is not broken (=classes inheriting from DeserializationContext won't break if they don't provide implementation for all methods).
So I changed the method to virtual with default implementation mostly for consistency with the methods that will be added in the future.